src/core/ui.js

```diff
@@ -79,6 +82,44 @@ document.documentElement.addEventListener("click", () => {
 });
 respecUI.appendChild(menu);
 
+menu.addEventListener("keydown", e => {
+  if (e.key === "Escape" && !menu.hidden) {
+    menu.classList.add("respec-hidden");
+    menu.classList.remove("respec-visible");
+    respecPill.setAttribute("aria-expanded", String(menu.hidden));
+    respecPill.focus();
+    menu.hidden = true;
+  }
+});
+
+// Code adapted from https://hiddedevries.nl/en/blog/2017-01-29-using-javascript-to-trap-focus-in-an-element
+function trapFocus(element) {
+  const focusableEls = element.querySelectorAll(
+    'a[href]:not([disabled]), button:not([disabled]), textarea:not([disabled]), input[type="text"]:not([disabled]), input[type="radio"]:not([disabled]), input[type="checkbox"]:not([disabled]), select:not([disabled])'
+  );
+  const firstFocusableEl = focusableEls[0];
+  const lastFocusableEl = focusableEls[focusableEls.length - 1];
+
+  element.addEventListener("keydown", e => {
+    if (e.key !== "Tab") {
+      return;
+    }
+    // shift + tab
+    if (e.shiftKey) {
+      if (document.activeElement === firstFocusableEl) {
+        lastFocusableEl.focus();
+        e.preventDefault();
+      }
+      return;
```

Comment:
switch statement may simplify this 👀

Reply:
what would we switch on? do you mean for `key`?

Reply:
Oops, misread, sorry 🤦‍♀️

Still I think else-if is better than return + if.

Reply:
I would still want to see this rewritten as else-if 👀

Reply:
`return` is redundant